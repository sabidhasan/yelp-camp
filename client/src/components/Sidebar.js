import React from 'react'
import PropTypes from 'prop-types'
import PricingTile from './PricingTile'

const Sidebar = (props) => {
  const paymentLogos = {'interac': 'ðŸ’³', cash: 'ðŸ’µ'};
  return (
    <aside
      className='Sidebar'
    >
      <section className='Sidebar__section'>
        <h2 className='Sidebar__heading'>Hours</h2>
        <div className='Sidebar__details'>
          {props.hours && props.hours.daily ?
            <React.Fragment>
              <h3 className='Sidebar__subheading'>Daily</h3>
              <div className='Sidebar__text'>{props.hours.daily}</div>
            </React.Fragment>
          : <div className='Sidebar__text'>No daily hours for this campground</div>}
          {props.hours && props.hours.seasonal.filter(v => !!v).length ?
            <React.Fragment>
              <h3 className='Sidebar__subheading'>Seasonal</h3>
              <div className='Sidebar__text'>{props.hours.seasonal.filter(v => !!v).join(' to ')}</div>
            </React.Fragment>
          : <div className='Sidebar__text'>No seasonal hours for this campground</div>}
        </div>
      </section>

      <section className='Sidebar__section'>
        <h2 className='Sidebar__heading'>Camp Sites</h2>
        <div className='Sidebar__details'>
          <div className='Sidebar__text'>
            {props.campsites || `No details on how many campsites at this location`}
          </div>
        </div>
      </section>

      <section className='Sidebar__section'>
        <h2 className='Sidebar__heading'>Prices</h2>
        <div className='Sidebar__details'>
          <h3 className='Sidebar__subheading'>Visitors</h3>
          <div className='Sidebar__text'>
            {props.prices ? (props.prices.visitors || 'Free')
            : 'No visitor pricing provided'}
          </div>
          <PricingTile type='daily' data={props} />
          <PricingTile type='weekly' data={props} />
          <PricingTile type='seasonal' data={props} />

          <div className='Sidebar__text'>{props.prices ? props.prices.description : ''}</div>
        </div>
      </section>

      <section className='Sidebar__section'>
        <h2 className='Sidebar__heading'>Payment Methods</h2>
        <div className='Sidebar__text'>
          {props.paymentMethods && props.paymentMethods.length ? props.paymentMethods.map((val, idx) => <h2 key={idx}>{paymentLogos[val]} {val}</h2>)
          : 'No payment method information found'}
        </div>
      </section>
    </aside>
  )
}

Sidebar.propTypes = {
    hours: PropTypes.object,
    campsites: PropTypes.number,
    prices: PropTypes.object,
    paymentMethods: PropTypes.array
}

export default Sidebar
