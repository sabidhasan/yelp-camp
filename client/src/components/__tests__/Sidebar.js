import React from 'react'
import toJson from 'enzyme-to-json'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Sidebar from '../Sidebar'

configure({ adapter: new Adapter() });

describe('<Sidebar />', () => {
  let component;
  let props;

  beforeEach(() => {
    props = {
      hours: {daily: [], seasonal: ['1', '2']},
      campsites: undefined,
      prices: {},
      paymentMethods: ['interac', 'cash', 'cash']
    }
    component = shallow(<Sidebar { ...props } />)
  });

  it('Component should be defined', () => {
    expect(component).toBeDefined();
  });

  it('Sidebar doesnt show daily hours if not given props', () => {
    component.setProps({hours: {daily: undefined, seasonal: []}})
    expect(component.find('.Sidebar__daily')).toHaveLength(0);
  });

  it('Sidebar doesnt show seasonal hours if no seasonal hours given', () => {
    component.setProps({hours: {daily: undefined, seasonal: []}})
    expect(component.find('.Sidebar__seasonal')).toHaveLength(0);
  });

  it('Component should be defined', () => {
    expect(component.find('.Sidebar__seasonal-text').text()).toBe('1 to 2');
  });

  it('Sidebar doesnt show seasonal hours if no seasonal hours given', () => {
    component.setProps({campsites: 20})
    expect(component.find('.Sidebar__campsites').text()).toBe('20');
  });

  it('If no visitor pricing information provided, default to `Free`', () => {
    expect(component.find('.Sidebar__pricing').text()).toBe('Free')
  })

  it('If visitor pricing information is provided then show pricing', () => {
    component.setProps({prices: {visitors: 10}})
    expect(component.find('.Sidebar__pricing').text()).toBe('10')
  })

  it('Number of payment methods matches what is provided via props', () => {
    expect(component.find('.Sidebar__payment')).toHaveLength(3)
  })

  it('LoginForm component snapshot matches what is expected', () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
