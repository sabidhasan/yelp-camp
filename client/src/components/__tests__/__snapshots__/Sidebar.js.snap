// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar /> LoginForm component snapshot matches what is expected 1`] = `
<aside
  className="Sidebar"
>
  <section
    className="Sidebar__section"
  >
    <h2
      className="Sidebar__heading"
    >
      Hours
    </h2>
    <div
      className="Sidebar__details"
    >
      <h3
        className="Sidebar__subheading Sidebar__daily"
      >
        Daily
      </h3>
      <div
        className="Sidebar__text"
      />
      <h3
        className="Sidebar__subheading Sidebar__seasonal"
      >
        Seasonal
      </h3>
      <div
        className="Sidebar__text Sidebar__seasonal-text"
      >
        1 to 2
      </div>
    </div>
  </section>
  <section
    className="Sidebar__section"
  >
    <h2
      className="Sidebar__heading"
    >
      Camp Sites
    </h2>
    <div
      className="Sidebar__details"
    >
      <div
        className="Sidebar__text Sidebar__campsites"
      >
        No details on how many campsites at this location
      </div>
    </div>
  </section>
  <section
    className="Sidebar__section"
  >
    <h2
      className="Sidebar__heading"
    >
      Prices
    </h2>
    <div
      className="Sidebar__details"
    >
      <h3
        className="Sidebar__subheading"
      >
        Visitors
      </h3>
      <div
        className="Sidebar__text Sidebar__pricing"
      >
        Free
      </div>
      <PricingTile
        data={
          Object {
            "campsites": undefined,
            "hours": Object {
              "daily": Array [],
              "seasonal": Array [
                "1",
                "2",
              ],
            },
            "paymentMethods": Array [
              "interac",
              "cash",
              "cash",
            ],
            "prices": Object {},
          }
        }
        type="daily"
      />
      <PricingTile
        data={
          Object {
            "campsites": undefined,
            "hours": Object {
              "daily": Array [],
              "seasonal": Array [
                "1",
                "2",
              ],
            },
            "paymentMethods": Array [
              "interac",
              "cash",
              "cash",
            ],
            "prices": Object {},
          }
        }
        type="weekly"
      />
      <PricingTile
        data={
          Object {
            "campsites": undefined,
            "hours": Object {
              "daily": Array [],
              "seasonal": Array [
                "1",
                "2",
              ],
            },
            "paymentMethods": Array [
              "interac",
              "cash",
              "cash",
            ],
            "prices": Object {},
          }
        }
        type="seasonal"
      />
      <div
        className="Sidebar__text"
      />
    </div>
  </section>
  <section
    className="Sidebar__section"
  >
    <h2
      className="Sidebar__heading"
    >
      Payment Methods
    </h2>
    <div
      className="Sidebar__text"
    >
      <h2
        className="Sidebar__payment"
        key="0"
      >
        ðŸ’³
         
        interac
      </h2>
      <h2
        className="Sidebar__payment"
        key="1"
      >
        ðŸ’µ
         
        cash
      </h2>
      <h2
        className="Sidebar__payment"
        key="2"
      >
        ðŸ’µ
         
        cash
      </h2>
    </div>
  </section>
</aside>
`;
